File: VB/DataLayer/Data/Classes/Category.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations.Schema

Namespace DataLayer.Data.Classes
    Public Class Category
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)> _
        Public Property ID() As Integer
        Public Property CategoryName() As String
        <NotMapped> _
        Public Property TransactionType() As TransactionType
            Get
                Return CType(TransactionTypeId, TransactionType)
            End Get
            Set(ByVal value As TransactionType)
                TransactionTypeId = CInt(value)
            End Set
        End Property
        Public Property TransactionTypeId() As Integer
        Public Property Description() As String
    End Class
End Namespace

File: CS/DataLayer/Data/Classes/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Data.Classes
{
    public class Category {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public string CategoryName { get; set; }
        [NotMapped]
        public TransactionType TransactionType { get { return (TransactionType)TransactionTypeId; } set { TransactionTypeId = (int)value; } }
        public int TransactionTypeId { get; set; }
        public string Description { get; set; }
    }
}
